### PR TITLE
#3943 - Bug - restriction view when access limited 

### PR DESCRIPTION
**Pull request by lewischen-aot**
- Removed the check-role-permission decorator to allow ministry users without designated roles to close the modal as per AC
  - The 'Close' button is not disabled for all users. Users without restriction edit access should be able to view it, not edit it and then close the modal without refreshing.

Screenshot
image

### CODE REVIEW

File: sources/packages/web/src/components/common/restriction/ViewRestriction.vue [hunk 1]
```diff
@@ -160,6 +153,8 @@ export default defineComponent({
       return "Resolution reason is required.";
     };
 
+    console.info("props.canResolveRestriction: ", props.canResolveRestriction);
```
Comment (dheepak-aot) [inline sources/packages/web/src/components/common/restriction/ViewRestriction.vue:156]: Please remove the console.info.

File: sources/packages/web/src/components/common/restriction/ViewRestriction.vue [hunk 2]
```diff
@@ -73,19 +73,14 @@
...
-              :primaryLabel="allowUserToEdit ? 'Resolve restriction' : 'Close'"
-              secondaryLabel="Cancel"
-              @primaryClick="allowUserToEdit ? submit() : cancel()"
-              @secondaryClick="cancel"
-              :disablePrimaryButton="notAllowed"
-              :showSecondaryButton="allowUserToEdit"
-            />
-          </template>
-        </check-permission-role>
+        <footer-buttons
```
Comment (dheepak-aot) [inline sources/packages/web/src/components/common/restriction/ViewRestriction.vue:76]: Check permission role is required and must not be removed.

File: sources/packages/web/src/components/common/restriction/ViewRestriction.vue [hunk 3]
```diff
@@ -73,19 +73,14 @@
...
+          :processing="processing"
+          :primaryLabel="allowUserToEdit ? 'Resolve restriction' : 'Close'"
```
Comment (dheepak-aot) [inline sources/packages/web/src/components/common/restriction/ViewRestriction.vue:78]: I would suggest the following quick solution which would keep the primary behavior as primary and secondary behaviour as secondary not having to juggle based on the situation.

**Solution**

```vue
<template #footer>
        <check-permission-role :role="allowedRole">
          <template #="{ notAllowed }">
            <footer-buttons
              :processing="processing"
              primaryLabel="Resolve restriction"
              :secondaryLabel="allowUserToEdit ? 'Cancel' : 'Close'"
              @primaryClick="submit"
              @secondaryClick="cancel"
              :disablePrimaryButton="notAllowed"
              :showPrimaryButton="allowUserToEdit"
            />
          </template>
        </check-permission-role>
      </template>
```

Result: 

View Active Restriction with permission

image


View resolved restriction with permission

image

View Resolved restriction without permission

image

File: sources/packages/web/src/components/common/restriction/ViewRestriction.vue [hunk 4]
```diff
@@ -73,19 +73,14 @@
...
+          secondaryLabel="Cancel"
+          @primaryClick="allowUserToEdit ? submit() : cancel()"
+          @secondaryClick="cancel"
+          :showSecondaryButton="allowUserToEdit"
```
Comment (dheepak-aot) [inline sources/packages/web/src/components/common/restriction/ViewRestriction.vue:82]: Or as a bare minimal solution. 

image

**Comment by dheepak-aot** [comment]
Please have a look at the comments.

File: sources/packages/web/src/components/common/restriction/ViewRestriction.vue (hunk 4 above)
Comment (lewischen-aot) [reply sources/packages/web/src/components/common/restriction/ViewRestriction.vue]: I agree with this solution, which is adopted in the latest commit.

**Review by dheepak-aot — APPROVED** [review]
Looks good @lewischen-aot 👍

**Review by andrepestana-aot — APPROVED** [review]
👍

File: sources/packages/web/src/components/common/restriction/ViewRestriction.vue (hunk 3 above)
Comment (lewischen-aot) [reply sources/packages/web/src/components/common/restriction/ViewRestriction.vue]: The solution looks great, and we can probably adopt this in other stories. Thanks @dheepak-aot

**Pull request merged**